autofit/non_linear/plot/samples_plotters.py

```diff
@@ -0,0 +1,120 @@
+import matplotlib.pyplot as plt
+import numpy as np
+from functools import wraps
+import logging
+import os
+
+from autofit.non_linear.plot.output import Output
+
+logger = logging.getLogger(__name__)
+
+def skip_plot_in_test_mode(func):
+    """
+    Skips visualization plots of non-linear searches if test mode is on.
+
+    Parameters
+    ----------
+    func
+        A function which plots a result of a non-linear search.
+
+    Returns
+    -------
+        A function that plots a visual, or None if test mode is on.
+    """
+
+    @wraps(func)
+    def wrapper(
+        obj: object,
+        *args,
+        **kwargs
+    ):

```

Comment:
If you're not using the obj argument explicitly then you can make a more versatile decorator with just *args and **kwargs